sequencing/batch_queue.go

```diff
@@ -0,0 +1,140 @@
+package sequencing
+
+import (
+	"sync"
+
+	"github.com/dgraph-io/badger/v4"
+	"github.com/rollkit/go-sequencing"
+)
+
+// BatchQueue ...
+type BatchQueue struct {
+	queue []sequencing.Batch
+	mu    sync.Mutex
+}
+
+// NewBatchQueue creates a new BatchQueue
+func NewBatchQueue() *BatchQueue {
+	return &BatchQueue{
+		queue: make([]sequencing.Batch, 0),
+	}
+}
+
+// AddBatch adds a new batch to the queue
+func (bq *BatchQueue) AddBatch(batch sequencing.Batch, db *badger.DB) error {
+	bq.mu.Lock()
+	bq.queue = append(bq.queue, batch)
+	bq.mu.Unlock()
+
```

Comment:
Wouldn't it make sense to perform the batch validation steps before adding the batch to the queue? If we can't Hash or Marshal the batch, I would think that we should reject it from the queue.